test/test_ops.py

```diff
@@ -171,6 +171,27 @@ def unsupported(dtype):
 
         self.assertEqual(supported_backward_dtypes, claimed_backward_supported, msg=msg)
 
+    # Validates that each OpInfo works correctly on different CUDA devices
+    @skipCUDAIfRocm
+    @onlyCUDA
+    @ops(op_db, allowed_dtypes=(torch.float32, torch.long))
+    def test_multiple_devices(self, device, dtype, op):
+        if torch.cuda.device_count() < 2:
+            self.skipTest("Skipped! Multiple CUDA devices are not available.")
+
+        available_cuda_devices = self.get_all_devices()
```

Comment:
What about setting the device to the first string representation in the list and then testing on the second string representation only? Would that better capture the original issue?

This should be wrapped in a try/finally to reset the device back to the device it was originally set to

Reply:
Do you think going through all available devices is too much for this test?

I think your suggestion with setting one device and test the other should be testing the same. So I can change it. Currently the device is implicitly set to cuda:0 (or maybe some other primary id).